### PR TITLE
feat(d.ts): improve axe.d.ts types

### PR DESCRIPTION
**Pull request by hamax**
Improve axe.d.ts types based on what we found was missing/incorrect and was needed in our code. I added more details for each change in commit messages (I tried to verify that each change is correct by looking at axe internals). Some of the issues were introduced with a recent update in #3966, which prompted me to look into this more.

### CODE REVIEW

File: axe.d.ts [hunk 1]
```diff
@@ -367,6 +391,34 @@ declare namespace axe {
...
+    ) => SerialDqElement;
+    uuid: (
+      options?: { random?: Uint8Array | Array<number> },
+      buf?: Uint8Array | Array<number>,
+      offset?: number
+    ) => string | Uint8Array | Array<number>;
+  }
+
+  interface Aria {
+    getRoleType: (role: string) => string | null;
```
Comment (WilcoFiers) [inline axe.d.ts:406]:
```suggestion
    getRoleType: (role: string | Element | VirtualNode | null) => string | null;
```

File: axe.d.ts [hunk 2]
```diff
@@ -367,6 +391,34 @@ declare namespace axe {
...
+  }
+
+  interface Dom {
+    isFocusable: (node: Element) => boolean;
+    isNativelyFocusable: (node: Element) => boolean;
```
Comment (WilcoFiers) [inline axe.d.ts:411]:
```suggestion
    isFocusable: (node: Element | VirtualNode) => boolean;
    isNativelyFocusable: (node: Element | VirtualNode) => boolean;
```

File: axe.d.ts [hunk 3]
```diff
@@ -367,6 +391,34 @@ declare namespace axe {
...
+  }
+
+  interface Text {
+    accessibleText: (element: Element, inLabelledByContext?: boolean) => string;
```
Comment (WilcoFiers) [inline axe.d.ts:415]:
```suggestion

  type AccessibleTextOptions = {
    inControlContext?: boolean
    inLabelledByContext?: boolean
  }
  
  interface Text {
    accessibleText: (element: Element, options?: AccessibleTextOptions) => string;
```

File: axe.d.ts [hunk 4]
```diff
@@ -257,10 +260,30 @@ declare namespace axe {
     brand?: string;
     application?: string;
   }
+  interface CheckHelper {
+    async: () => (result: boolean | undefined | Error) => void;
+    data: (data: unknown) => void;
+    relatedNodes: (nodes: Element[]) => void;
+  }
+  interface EvaluateResult {
```
Comment (WilcoFiers) [inline axe.d.ts:268]: This is what we've been calling this internally:

```suggestion
  interface AfterResult {
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
Thanks for opening this PR. I am really curious what you are using these methods for.

**Comment by hamax** [comment]
Thanks, I amended the commits with suggestions (+ prettier formatting).

> Thanks for opening this PR. I am really curious what you are using these methods for.

Mostly for custom rules/checks and capturing additional debug information for nodes in results. I think it makes sense for utils commonly used from checks to be available.

**Review by WilcoFiers — APPROVED** [review]

**Comment by WilcoFiers** [comment]
✅  Reviewed for security.

**Comment by WilcoFiers** [comment]
Thank you again for your contribution @hamax!

**Pull request merged**